index.js

```diff
@@ -0,0 +1,46 @@
+function calculate(arr) {
+  const operands = []
+  let operation = null
+  const findOperation = ['+', '-', '*', '/']
+
+
+  for (let i = 0; i < arr.length; i++) {
+    if (Number.isInteger(arr[i]) || Number.isInteger(parseInt(arr[i]))) {
+      operands.push(arr[i])
+    }
+    else if (arr[i] === null) {
+      operands.push(0)
+    }
+    else if (isOperation(findOperation, arr[i])) {
+      operation = arr[i]
+    }
+  }
+
+  if (findOperation.indexOf(operation) === -1) {
+    return NaN
+  }
```

Comment:
You're looping twice here when you don't need to. See the solution branch, these can all be done in one step.